### PR TITLE
Emit -points does not add time

### PR DESCRIPTION
**Pull request by bkutil**

### CODE REVIEW

**Comment by bkutil** [comment]
cc: @dmehra @go-oleg @dmajda

**Comment by bkutil** [comment]
refs: #77

**Comment by dmehra** [comment]
Does this also cover "not emitting ticks and emitting all points at once" as per last comments in #77?

**Comment by bkutil** [comment]
Yes, updated the commit message.

**Comment by demmer** [comment]
This looks good, though I think we want to add a check that the user shouldn't use `-points` with any of `-from, -to, -last, -limit, -every` since they'll now be ignored.

**Comment by demmer** [comment]
Also given this new behavior, you should modify `next_point` so it doesn't have the check for `this.points` since that'll never happen any more.

**Comment by bkutil** [comment]
> This looks good, though I think we want to add a check that the user shouldn't use -points with any of -from, -to, -last, -limit, -every since they'll now be ignored.

Ok, I added the missing checks and fixed the tests.

> Also given this new behavior, you should modify next_point so it doesn't have the check for this.points since that'll never happen any more.

Won't `this.points` check still be needed when we iterate over `-points` containiing time?

File: lib/runtime/procs/emit.js [hunk 1]
```diff
@@ -54,23 +54,30 @@ var emit = source.extend({
             throw this.compile_error('RT-EMIT-TO-LIMIT-ERROR');
         }
         if (_.has(options, 'points')) {
+            // Invalid options
             if (_.has(options, 'to') || _.has(options, 'limit')) {
                 throw this.compile_error('RT-EMIT-POINTS-LIMIT-ERROR');
             }
+            if (_.has(options, 'from') || _.has(options, 'every')) {
```
Comment (demmer) [inline lib/runtime/procs/emit.js]: What about `-to`?

**Comment by demmer** [comment]
> Won't this.points check still be needed when we iterate over -points containing time?

Oh of course -- I misread the PR. You're correct.

Modulo the missing check for `-to` I think this looks good. We will have a conflict once #248 merges since I reworked a bit of the internals of `emit`, but that should be easy enough to resolve.

File: lib/runtime/procs/emit.js (hunk 1 above)
Comment (bkutil) [inline lib/runtime/procs/emit.js]: Three lines above this one? :)

**Pull request merged**